libsplinter/src/oauth/rest_api/actix/callback.rs

```diff
@@ -98,6 +106,27 @@ pub fn make_callback_route(
                                             HttpResponse::InternalServerError()
                                                 .json(ErrorResponse::internal_error())
                                         }
+                                    };
+                                    #[cfg(feature = "biome-profile")]
+                                    {
+                                        match save_user_profile(
+                                            user_profile_store.clone_box(),
+                                            oauth_user_session_store.clone_box(),
+                                            &user_info.profile().clone(),
+                                            user_info.subject.clone(),
+                                        ) {
+                                            Ok(_) => {
+                                                debug!("User profile saved");
+                                                HttpResponse::Found()
+                                                    .header(LOCATION, redirect_url)
+                                                    .finish()
+                                            }
+                                            Err(err) => {
+                                                error!("Failed to save profile: {}", err);
+                                                HttpResponse::InternalServerError()
+                                                    .json(ErrorResponse::internal_error())
+                                            }
```

Comment:
The `HttpResponse::Found` response should be outside of this match and the `HttpResponse::InternalServerError` should be explicitly `return`ed. If the `biome-profile` feature isn't enabled, this code won't compile because there won't be anything returning an `HttpResponse`.

Reply:
Fixed